### PR TITLE
FHIR Convert API *Preview* documentation

### PR DESCRIPTION
**Pull request by dustinburson**
Initial documentation release FHIR Converter API *Preview*

- Update main readme for new updated container based convert.
- Readme updated with previous version section, which directs to prior versions.
- Create concept documents for resource id generation and validation.
- Add example using built in date filter to Filter and Tags documentation.
- Remove no longer relevant TempleteManagementCLI.md from main.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,215 +1,93 @@
 # FHIR Converter
 
-FHIR Converter is an open source project that enables conversion of health data from legacy formats to FHIR.
+FHIR Converter is an open source project that enables conversion of health data from legacy formats to and from FHIR.  The FHIR Converter uses the [Liquid template language](https://shopify.github.io/liquid/) and the .NET runtime.
```
Comment (pallar-ms) [inline README.md:3]: super nit: I believe Linishya wanted the 'c' to be lowercase.

File: README.md [hunk 2]
```diff
@@ -1,215 +1,93 @@
...
 
-The converter comes with a few ready-to-use templates. If needed, you can create a new template, or modify existing templates to meet your specific conversion requirements. The provided templates are based off of HL7 v2.8. Other versions may require you to make modifications to these templates on your own. 
+## Architecture
 
-FHIR Converter with DotLiquid engine transforms the input data into FHIR bundles that are persisted to a FHIR server. The converter is integrated into both [Azure Health Data Services](https://azure.microsoft.com/en-us/services/health-data-services/#overview) and [Azure API for FHIR](https://docs.microsoft.com/azure/healthcare-apis/azure-api-for-fhir/), as well as in the open-source [FHIR Server](https://github.com/microsoft/fhir-server) as a [`$convert-data`] #using-convert-data-in-fhir-server) operation. It is also available as a [command line tool](#using-command-line-tool).
+The FHIR converter enables healthcare record format conversion scenarios from various source formats (such as Hl7v2, CCDA, JSON, FHIR STU3) to destination formats (such as FHIR R4) using Liquid templates (to specify the transformation rules to be applied).
```
Comment (pallar-ms) [inline README.md:24]: nit: This feels redundant with the first line in this readme. Can be removed?

File: README.md [hunk 3]
```diff
@@ -1,215 +1,93 @@
...
-        }
+## Troubleshooting
 
-        return false;
-    }
+Some key concepts to consider:
+* Processing time is related to both the input message size, template, and logic contained in the template.  If your template is taking a long time to execute make sure you don't have any unnecessary loops.
+* The output of the template is expected to be JSON when the target is FHIR.
+* When converting data to FHIR, [post processing](https://github.com/microsoft/FHIR-Converter/blob/main/src/Microsoft.Health.Fhir.Liquid.Converter/OutputProcessors/PostProcessor.cs) is performed.  If you are seeing unexpected results, double check the post processing logic. 
+* If you want a deeper understanding on how data is converter, look at the functional tests found [here](https://github.com/microsoft/FHIR-Converter/blob/main/src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/ConvertDataTemplateCollectionProviderFunctionalTests.cs)
```
Comment (pallar-ms) [inline README.md:75]: nit: converted*

**Review by pallar-ms — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,215 +1,93 @@
...
-## Using Command line tool
+The FHIR Converter API comes with several pre-built templates you can use as reference as to create your own.
 
-### Supported parameters
+| Conversion | Notes |
+| ----- | ----- |
+| [HL7v2 to FHIR](/docs/HL7v2-templates.md)| Important points to note for HL7v2 to FHIR conversion: [see here](docs/HL7v2-ImportantPoints.md) <br> Common FHIR Validator errors/warning you might run into, and their explanations: [see here](docs/HL7v2-FHIRValidator.md) | 
+| [C-CDA to FHIR](/data/Templates/Ccda/) | | 
+| [JSON to FHIR](/data/Templates/Json/) | | 
+| [FHIR STU3 to R4](/data/Templates/Stu3ToR4/) | [Diferences between STU3 & R4](/docs/Stu3R4-resources-differences.md) |
```
Comment (ShaunDonn2) [inline README.md:44]: nit: spelling "differences"

File: README.md (hunk 2 above)
Comment (dustinburson) [reply README.md:24]: Sure I will look at some new wording/removing.

File: README.md (hunk 1 above)
Comment (dustinburson) [reply README.md:3]: Okay, I can make that change.  It doesn't align with the repo name.

**Review by ShaunDonn2 — DISMISSED** [review]

**Review by wi-y — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (dustinburson) [reply README.md]: Updated

File: README.md (hunk 2 above)
Comment (dustinburson) [reply README.md]: Updated

File: README.md (hunk 3 above)
Comment (dustinburson) [reply README.md]: Fixed

File: README.md [hunk 5]
```diff
@@ -1,215 +1,93 @@
...
 
-|  | Handlebars engine | Liquid engine | 
-| ----- | ----- | ----- |
-| **Template language** | [Handlebars](https://handlebarsjs.com/) | [Liquid](https://shopify.github.io/liquid/) |
-| **Template authoring tool** | Self-hosted web-app | [VS Code extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-health-fhir-converter)|
-| **Supported conversions** | 1. HL7v2 to FHIR <br> 2. C-CDA to FHIR | 1. HL7v2 to FHIR <br> 2. C-CDA to FHIR <br> 3. JSON to FHIR <br> 4. FHIR STU3 to FHIR R4 **(new!)** |
-| **Available as** | 1. Self-deployed web service <br> (on-prem or on Azure)| 1. Command line tool <br> 2. $convert-data operation in  [FHIR Server for Azure (OSS)](https://github.com/microsoft/fhir-server/blob/main/docs/ConvertDataOperation.md) <br> 3. $convert-data operation in both [Azure Health Data Services](https://docs.microsoft.com/en-us/azure/healthcare-apis/fhir/convert-data) and [Azure API for FHIR](https://docs.microsoft.com/azure/healthcare-apis/azure-api-for-fhir/convert-data)|
+## What's New?
+The latest iteration of the *Preview* FHIR Converter makes some sigifigant changes over [previous versions](#previous-versions).
```
[the remark this answers is not in the document]
Comment (dustinburson) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (dustinburson) [reply README.md]: Fixed

**Review by pallar-ms — APPROVED** [review]

**Review by ShaunDonn2 — APPROVED** [review]

**Pull request merged**